server.js

```diff
@@ -1,29 +1,67 @@
-const http = require("http") 
-const fs = require("fs/promises")
+const http = require("http");
+const fs = require("fs/promises");
 
-const getFiles = async ()=>{
-    const indexHtml = await fs.readFile("./index.html")
-    const indexJs = await fs.readFile("./dist/r.bundle.js")
-    return [indexHtml,indexJs]
-}
+const getFiles = async () => {
+    const indexHtml = await fs.readFile("./index.html");
+    const indexJs = await fs.readFile("./dist/r.bundle.js");
+    return [indexHtml, indexJs];
+};
 
+const registrations = []
 
 http.createServer(async ({ url }, res) => {
-    const [indexHtml, indexJs] = await getFiles()
-    switch(url) {
-        case '/':
-            res.writeHead(200, {"Content-Type": "text/html"})
-            res.write(indexHtml)
-            res.end()
-            break
-        case '/js':
-            res.writeHead(200, {"Content-Type": "text/javascript"})
-            res.write(indexJs);
-            res.end();      
-            break
-        default:
-            res.writeHead(404)
-            res.write('Page not found');
-            res.end();      
+    const [indexHtml, indexJs] = await getFiles();
+    if (url.includes("Registration")) {
+        const registrationObjectUrl = url.split('=')[1];
+        const decodedReg = JSON.parse(decodeURIComponent(registrationObjectUrl));
+        const exists = registrations.some(registration => 
+            registration.email === decodedReg.email)
+        if (!exists) {
+            registrations.push(decodedReg);
+        }
+        const response = exists ? JSON.stringify(decodedReg) : exists.toString();
+        const buffer = Buffer.from(response);
+        console.log("reg",registrations)
+        res.writeHead(200, { "Content-Type": "text/javascript" });
+        res.write(buffer);
+        res.end();
     }
-}).listen(8008)
\ No newline at end of file
+    else if (url.includes("youLogIn")) {
+        const LoginObjectUrl = url.split('=')[1];
+        const decodedlog = JSON.parse(decodeURIComponent(LoginObjectUrl));
+        const stringify = JSON.stringify(decodedlog);
+        let loggedIn = false;
```

Comment:
This variable should be kept outside the condition (preferably in the same scope as the registrations array). At the beginning  it should be set to `false`, when we login it should be changed to `true` and when we logout it should be set to `false`.